provider/provider.go

```diff
@@ -102,6 +102,34 @@ func (p *Provider) ConfigureProvider(ctx context.Context, request *cqproto.Confi
 	return &cqproto.ConfigureProviderResponse{}, nil
 }
 
+func (p *Provider) CheckDuplicates() error {
+	tables := make(map[string]string)
+
+	// getTableNames function for recursive collection of table names from shcema.Table
+	var getTableNames func(resource *schema.Table) []string
```

Comment:
why not create a map[string]bool, and run recursively once on all tables and relations i.e ScanTableDuplicates(table, map[string]bool)  bool and execute this function in a for loop for each table in resource map? feels like shorter code.